app-shared/src/iosMain/kotlin/io/github/droidkaigi/confsched/IosExternalNavController.kt

```diff
@@ -34,11 +34,25 @@ actual fun rememberExternalNavController(): ExternalNavController {
 internal class IosExternalNavController : ExternalNavController {
     override fun navigate(url: String) {
         val nsUrl = NSURL.URLWithString(url)
-        if (nsUrl != null) {
-            UIApplication.sharedApplication.openURL(nsUrl)
-        } else {
-            println("Failed to navigate to URL: $url")
+        if (nsUrl == null) {
+            println("Failed to navigate to URL (invalid URL): $url")
+            return
         }
+
+        if (!UIApplication.sharedApplication.canOpenURL(nsUrl)) {
+            println("Cannot open URL (scheme not allowed or no handler): $url")
+            return
+        }
+
+        UIApplication.sharedApplication.openURL(
```

Comment:
Since the minimum deployment is iOS 18.4, we determined that fallback processing is unnecessary.